lychee-lib/src/client.rs

```diff
@@ -524,10 +536,18 @@ impl Client {
         uri: &Uri,
         credentials: &Option<BasicAuthCredentials>,
     ) -> Result<Status> {
-        match self.check_website_inner(uri, credentials).await {
+        let quirks = Quirks::default();
+        let mut request_chain: RequestChain = Chain::new(vec![Box::new(quirks)]);
+        request_chain.append(&mut self.request_chain.lock().unwrap());
+
+        if let Some(c) = credentials {
+            request_chain.push(Box::new(c.clone()));
+        }
+
+        match self.check_website_inner(uri, &mut request_chain).await {
             Status::Ok(code) if self.require_https && uri.scheme() == "http" => {
                 if self
-                    .check_website_inner(&uri.to_https()?, credentials)
+                    .check_website_inner(&uri.to_https()?, &mut request_chain)
```

Comment:
Since `request_chain` is a member of `Client`, do we have to pass it explicitly to `check_website_inner` and the other methods below? Shouldn't it be possible to use `self.request_chain` directly?

Reply:
I made `Client::request_chain` to be an empty chain by default and the chain added after the "internal" chain which consists of the `Quirks` and `BasicAuthCredentials` chain.

I think did this because `credentials` is passed to `check_website` as an argument, so I need to modify the chain then for this check but I don't want to change `self.request_chain` otherwise subsequent calls to `check_website` include the credentials.

It would be nice if there was just the one chain in `Client`. I can configure a default value where `Quirks` is included. Then I would need to remove `credentials` from `check_website` and users have to change the `request_chain` in order to use credentials. What do you think? Another advantage is that the full chain is fully configurable, currently quirks cannot be removed.

Reply:
As discussed it still is desirable to have a lychee-internal chain and the configurable plugin chain. I've created the helper structure `ClientRequestChain` to "merge" and handle multiple chains